src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -113,6 +113,11 @@ public class ProjectRestoreMetadata : IEquatable<ProjectRestoreMetadata>
 
         public RestoreLockProperties RestoreLockProperties { get; set; } = new RestoreLockProperties();
 
+        /// <summary>
+        /// Full path to the packages.config file, if it exists. Only valid when ProjectStyle is PackagesConfig.
+        /// </summary>
+        public string PackagesConfigPath { get; set; }
```

Comment:
This model is PackageReference/project.json based 😢

Reply:
It's used by PackageSpecWriter, and admittedly NuGet.targets previously considered packages.config projects as "unknown" and I changed it to specify the project type, but to me this feels more correct. Neither ProjectRestoreMetadata nor PackageSpecWriter, or the concept of the dependency graph feel PR/project.json specific.

Reply:
> Neither ProjectRestoreMetadata nor PackageSpecWriter, or the concept of the dependency graph feel PR/project.json specific.

I think the fact that they are used only in PR/project.json makes it PR/project.json specific by default. 
The PackageSpec model was created to represent project.json originally.

Reply:
then why does it return unknown projects? if it's meant to be PR/PJ specific, shouldn't it only return those project types and pretend that all the unknown project types don't exist? Not only does NuGet.targets flag them as unknown project types with a small amount of additional metadata, but the C# model has a class for these unknown projects, despite never being used. I understand it was created for PJ and used in PR since it works so similarly, but it feels incomplete to return unknown projects and pretend that PC doesn't exist.

Reply:
This merits more investigation/discussion.